test/Microsoft.AspNetCore.SignalR.Tests/HubEndpointTests.cs

```diff
@@ -36,6 +39,298 @@ public class HubEndpointTests
             }
         }
 
+        [Fact]
+        public async Task HubMethodCanReturnValueFromTask()
+        {
+            var serviceProvider = CreateServiceProvider();
+
+            var endPoint = serviceProvider.GetService<HubEndPoint<MethodHub>>();
+
+            using (var connectionWrapper = new ConnectionWrapper())
+            {
+                var endPointTask = endPoint.OnConnectedAsync(connectionWrapper.Connection);
+
+                await connectionWrapper.HttpConnection.Input.ReadingStarted;
+
+                var invocationAdapter = serviceProvider.GetService<InvocationAdapterRegistry>();
+                var writer = invocationAdapter.GetInvocationAdapter("json");
+                var serializer = new JsonSerializer();
```

Comment:
Can't you create this serializer in the ReadConnectionOutputAsync? Each test is polluted with this serializer even though it it not being used outside of ReadConnectionOutputAsync. You can't also really pass anything else to ReadConnectionOutputAsync.